src/ICSharpCode.SharpZipLib/GZip/GzipOutputStream.cs

```diff
@@ -208,7 +231,18 @@ void WriteHeader()
 					// The OS type (unknown)
 					(byte) 255
 				};
+
 				baseOutputStream_.Write(gzipHeader, 0, gzipHeader.Length);
+
+				//AFONSO DUTRA
+				if (!string.IsNullOrEmpty(_fileName))
+			    {
+			        var fileNameBytes = System.Text.Encoding.GetEncoding("ISO-8859-1").GetBytes(_fileName);
+			        Array.Resize<byte>(ref fileNameBytes, fileNameBytes.Length + 1);
+			        fileNameBytes[fileNameBytes.Length - 1] = 0;
+			        baseOutputStream_.Write(fileNameBytes, 0, fileNameBytes.Length);
```

Comment:
Perhaps nitpick, but you could simply call Write a second time with just a 0-byte instead of resizing the array.